datalad/runner/utils.py

```diff
@@ -87,7 +103,9 @@ def process(self, data: str) -> List[str]:
             # unterminated line.
             del detected_lines[-1]
 
-        return detected_lines
+            if self.keep_ends:
+                return [line + self.separator for line in detected_lines]
```

Comment:
how do we know that the last line did have `self.separator` -- don't we add the remainder as-is so it might have actually lacked the trailing separator?

Reply:
ah, that is probably what that exact logic+comment says above -- we keep it stored so lines were guaranteed to be terminated and non-terminated would be stored in `self.remaining_data` (not sure what its life span really...) and as such this should not change behavior, so ok